clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -322,7 +322,9 @@ void runOnce() {
                     client.poll(retryBackoffMs, time.milliseconds());
                     return;
                 } else if (transactionManager.hasAbortableError()) {
-                    accumulator.abortUndrainedBatches(transactionManager.lastError());
+                    if (accumulator.abortUndrainedBatches(transactionManager.lastError())) {
```

Comment:
Could you point me where do we eventually cancel the commit? Since it is not in diff I bet it happens somewhere later in the call trace but did not get it.

Reply:
Once the abortable error state has been reached, we rely on the user to abort the transaction.